spring-vault-core/src/main/java/org/springframework/vault/authentication/KubernetesAuthentication.java

```diff
@@ -105,8 +106,8 @@ public VaultToken login() throws VaultException {
 			return LoginTokenUtil.from(response.getAuth());
 		}
 		catch (HttpStatusCodeException e) {
-			throw new VaultException(String.format("Cannot login using kubernetes: %s",
-					VaultResponses.getError(e.getResponseBodyAsString())));
+			throw new VaultHttpException(String.format("Cannot login using kubernetes: %s",
```

Comment:
See above.

Reply:
I'm really going to need clarification of these "see above". Remember this is the only one exception fits all and further subclasses can clarify. Because login isn't the only problem.

Reply:
Was meant in the context of login to represent all top-level login failures within a `VaultLoginException`. Nested calls to external resources would retain `VaultHttpException`. Failures of calling `auth/{mount}/login` via HTTP or exceptions that bubble up otherwise would be wrapped within `VaultLoginException`.